internal/service/mysql/mysql_recovery.go

```diff
@@ -358,8 +380,8 @@ func GetMysqlRecovery(ctx context.Context, config *conn.ProviderConfig, no strin
 	serverList := resp.CloudMysqlInstanceList[0].CloudMysqlServerInstanceList
 
 	for _, server := range serverList {
-		if *server.CloudMysqlServerRole.CodeName == "Recovery" {
-			return server, nil
+		if (server.CloudMysqlServerRole != nil && *server.CloudMysqlServerRole.Code == "R") && (*server.CloudMysqlServerInstanceNo == serverInstanceNo) {
+			return []*vmysql.CloudMysqlServerInstance{server}, nil
```

Comment:
(trivial) GetMysqlRecovery() seems like returning a single `CloudMysqlServerInstance`, return type should be a `[]*vmysql.CloudMysqlServerInstance`?